Source/Core/VideoCommon/TextureConfig.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "VideoCommon/TextureConfig.h"
+
+#include <tuple>
+
+bool TextureConfig::operator==(const TextureConfig& o) const
```

Comment:
Did you see this function in profile? We have quite big maps based on this struct, so it might be better to move this implementation in the header.